dhis-2/dhis-services/dhis-service-node/src/main/java/org/hisp/dhis/fieldfilter/DefaultFieldFilterService.java

```diff
@@ -220,7 +220,7 @@ public CollectionNode toCollectionNode( Class<?> wrapper, FieldFilterParams para
                 .collect( Collectors.joining( "," ) );
         }
 
-        if ( params.getObjects().isEmpty() )
+        if ( params.getObjects().isEmpty() || objects.stream().noneMatch( Objects::nonNull ) )
```

Comment:
I think the double negation is difficult to read, none of the elements are not null

```suggestion
        if ( objects.isEmpty() || objects.stream().allMatch( Objects::isNull) )
```

Reply:
resolved